File: protocol/src/test/resources/luaTest/LuaProtocol/Packet/ObjectB.lua
-- @author jaysunxiao
-- @version 1.0
-- @since 2017 10.12 15:39

local ObjectB = {}

function ObjectB:new(flag)
    local obj = {
        flag = flag -- boolean
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function ObjectB:protocolId()
    return 1117
end

function ObjectB:write(byteBuffer, packet)
    if packet == null then
        byteBuffer:writeBoolean(false)
        return
    end
    byteBuffer:writeBoolean(true)
    byteBuffer:writeBoolean(packet.flag)
end

function ObjectB:read(byteBuffer)
    if not(byteBuffer:readBoolean()) then
        return nil
    end
    local packet = ObjectB:new()
    local result0 = byteBuffer:readBoolean()
    packet.flag = result0
    return packet
end

return ObjectB

File: protocol/src/test/resources/jsTest/jsProtocol/packet/ObjectB.js
// @author jaysunxiao
// @version 3.0
const ObjectB = function(flag) {
    this.flag = flag; // boolean
};

ObjectB.prototype.protocolId = function() {
    return 103;
};

ObjectB.write = function(byteBuffer, packet) {
    if (byteBuffer.writePacketFlag(packet)) {
        return;
    }
    byteBuffer.writeBoolean(packet.flag);
};

ObjectB.read = function(byteBuffer) {
    if (!byteBuffer.readBoolean()) {
        return null;
    }
    const packet = new ObjectB();
    const result0 = byteBuffer.readBoolean(); 
    packet.flag = result0;
    return packet;
};

export default ObjectB;
